bgpd/bgp_main.c

```diff
@@ -425,17 +425,21 @@ int main(int argc, char **argv)
 			else
 				bgp_port = tmp_port;
 			break;
-		case 'e':
-			multipath_num = atoi(optarg);
-			if (multipath_num > MULTIPATH_NUM
-			    || multipath_num <= 0) {
+		case 'e': {
+			unsigned long int parsed_multipath =
+				strtoul(optarg, NULL, 10);
+			if (parsed_multipath == 0
+			    || parsed_multipath > MULTIPATH_NUM
```

Comment:
MULTIPATH_NUM is 64 currently and may never exceed UINT_MAX. Given that shouldn't parsed_multipath > MULTIPATH_NUM covers for the parsed_multipath > UINT_MAX as well. So why do we explicitly need parsed_multipath > UINT_MAX condition ? Am I missing something ?

Reply:
>MULTIPATH_NUM is 64 and may never exceed UINT_MAX
It is 64 by default, and can be configured up to 999 if I read `configure.ac` correctly.

>Given that shouldn't parsed_multipath > MULTIPATH_NUM covers for the parsed_multipath > UINT_MAX as well. So why do we explicitly need parsed_multipath > UINT_MAX condition ?

We are assigning an `unsigned long` to an `unsigned int`. What harm does it do to make sure we aren't truncating? I see your point - assuming `configure.ac` is unmodified, then this is a dead check. On the other hand, including it makes the expected range explicit. What is your objection in particular?

Reply:
`strtoul()` goes up to `ULONG_MAX`, which is 64-bit while `UINT_MAX` is 32-bit.  The code is correct. (There is no `strtoui()`.)